### PR TITLE
docs: add output.publicPath: 'auto'

### PR DESCRIPTION
**Pull request by snitin315**
fixes https://github.com/webpack/webpack.js.org/issues/4491

add `output.publicPath: 'auto'`

### CODE REVIEW

File: src/content/configuration/output.mdx [hunk 1]
```diff
@@ -1135,6 +1135,7 @@ module.exports = {
   //...
   output: {
     // One of the below
+    publicPath: 'auto', // It automatically determines the publicPath from <script> tag. Only works with web and web-worker targets.
```
Comment (chenxsan) [inline src/content/configuration/output.mdx:1138]: `publicPath` can be derived in many ways more than `<script />` when set to `auto`. Here's the pull request introducing `auto` value for `publicPath`. https://github.com/webpack/webpack/pull/11258

File: src/content/guides/public-path.md [hunk 2]
```diff
@@ -59,3 +59,5 @@ W> Be aware that if you use ES6 module imports in your entry file the `__webpack
 import './public-path';
 import './app';
 ```
+
+T> You can also set `publicPath: 'auto'`. It automatically determines the publicPath from `import.meta.url`, `document.currentScript`, `<script/>` or `self.location`. It only works with `web` and `web-worker` targets.
```
Comment (chenxsan) [inline src/content/guides/public-path.md:63]:
```suggestion
T> When using `web` or `web-worker` target, `publicPath` defaults to `'auto'` which will automatically determines the public path from either `import.meta.url`, `document.currentScript`, `<script />` or `self.location`.
```

File: src/content/configuration/output.mdx [hunk 3]
```diff
@@ -1155,6 +1156,7 @@ __webpack_public_path__ = myRuntimePublicPath;
 
 See [this discussion](https://github.com/webpack/webpack/issues/2776#issuecomment-233208623) for more information on `__webpack_public_path__`.
 
+T> output.publicPath defaults to `'auto'` with `web` and `web-worker` targets.
```
Comment (chenxsan) [inline src/content/configuration/output.mdx:1159]:
```suggestion
T> `output.publicPath` defaults to `'auto'` with `web` and `web-worker` targets.
```

File: src/content/configuration/output.mdx [hunk 4]
```diff
@@ -1135,6 +1135,7 @@ module.exports = {
   //...
   output: {
     // One of the below
+    publicPath: 'auto', // It automatically determines the publicPath from `import.meta.url`, `document.currentScript`, `<script/>` or `self.location`.
```
Comment (chenxsan) [inline src/content/configuration/output.mdx:1138]:
```suggestion
    publicPath: 'auto', // It automatically determines the public path from either `import.meta.url`, `document.currentScript`, `<script />` or `self.location`.
```

File: src/content/guides/public-path.md [hunk 5]
```diff
@@ -59,3 +59,5 @@ W> Be aware that if you use ES6 module imports in your entry file the `__webpack
 import './public-path';
 import './app';
 ```
+
+T> When using `web` or `web-worker` target, `publicPath` defaults to `'auto'` which will automatically determines the public path from either `import.meta.url`, `document.currentScript`, `<script />` or `self.location`.
```
Comment (snitin315) [inline src/content/guides/public-path.md:63]:
```suggestion
T> When using `web` or `web-worker` target, `publicPath` defaults to `'auto'` which will automatically determine the public path from either `import.meta.url`, `document.currentScript`, `<script />` or `self.location`.
```

**Review by chenxsan — APPROVED** [review]

**Pull request merged**